src/storage/__tests__/storage.service.spec.ts

```diff
@@ -0,0 +1,134 @@
+                        get: jest.fn(() => 'src/storage/__tests__/test'),
+                    },
+                },
+            ],
+        }).compile();
+
+        service = module.get<StorageService>(StorageService);
+        config = module.get<ConfigService>(ConfigService);
+        filter = module.get<MainExceptionFilter>(MainExceptionFilter);
+
+        jest.clearAllMocks();
+    });
+
+    afterAll(() => {
+        mock.restore();
+    });
+
+    describe('readFile', () => {
+        it('should return the content of the file', async () => {
+            const fileName = 'test_id';
+            const contentInFile = 'Hello World';
+            mock({
+                'src/storage/__tests__/test': {
+                    [fileName]: contentInFile,
+                },
+            });
+
+            const readStream = await service.readFile(fileName);
+
+            expect(readStream).toEqual(contentInFile);
+        });
+
+        it('should throw error, when file not existing', async () => {
+            const fileName = 'notExistingFile.html';
+            mock({
+                'src/storage/__tests__/test': {},
+            });
+
+            //then
+            try {
+                await service.readFile(fileName);
+            } catch (err) {
+                filter.catch(err, {
+                    switchToHttp: () => ({
+                        getRequest: () => mockRequest,
+                        getResponse: () => mockResponse,
+                    }),
+                } as any);
+
+                expect(mockResponse.status).toHaveBeenCalledWith(404);
+                expect(mockResponse.json).toHaveBeenCalledWith({
+                    message: 'File does not exist',
+                });
+            }
+        });
+    });
+
+    describe('writeFile', () => {
+        it('should write file successfully', async () => {
+            // given
+            const fileId = 'test_id';
+            const file = {
+                mimetype: 'text/html',
+                buffer: Buffer.from('file contents'),
+            } as Express.Multer.File;
+
+            mock({
+                'src/storage/__tests__/test': {},
+            });
+
+            // when
+            const result = await service.saveFile(file);
+
+            // then
+            expect(result).toEqual(fileId);
+        });
+
+        it('should throw InternalServerErrorException when something goes wrong with writhing the file', async () => {
+            //given
+            const file = {
+                mimetype: 'text/html',
+                buffer: Buffer.from('file contents'),
+            } as Express.Multer.File;
+
+            jest.spyOn(service, 'saveFile').mockImplementation(() => {
+                throw new Error('Something goes wrong with saving the file');
+            });
+
+            //then
+            try {
+                await service.saveFile(file);
+            } catch (err) {
+                filter.catch(err, {
```

Comment:
w testach nie robi sie try/catch

Reply:
wiem wiem, bo jak nie wyskoczy błąd to przepuści test jak positive :)
tylko powiedz jak mam przetestować błąd, który łapie Nest?? bo jak rzucamy w service błąd domenowy to spoko, ale w przypadku innych błędów filter ma je wyłapywać, bez try catch to nie działa albo źle testy napisałem
Jak napisałem taki test na łapanie błędów jak dla domenowego to nie bangala :)

Reply:
to trzeba wtedy e2e test zrobic jak sie nie myle - latwiej bedzie jak ten test pomieniesz i manualnie to przetestujesz

Reply:
No właśnie przy takim błędzie raczej powinno się e2e zrobić, a przynajmniej tak znawcy ze stackoverflow napisali :)
Ręcznie to właśnie taki sposób znalazłem :) No  ale spróbuję...

Reply:
recznie to nie oznacza napisanie testu... recznie == przeklikaj za pomoca postmana endpoint/logike i sprawdz czy dziala